libs/transactions/src/lib/views/transactions-view/transactions-view.component.html

```diff
@@ -9,9 +9,18 @@ <h1 *ngIf="title !== ''" data-role="transactions-view__title">
       </div>
       <ul *ngIf="transactions$ | async as items; else loading" class="bb-list">
         <li
-          *ngFor="let item of items | filterTransactions: filter"
+          *ngFor="
+            let item of items | filterTransactions: filter:selectedAccount
+          "
           class="bb-list__item"
           data-role="transactions-view__item-container"
+          [routerLink]="['/transactions/transaction-details', item.id]"
+          [state]="{
+            queryParams: item,
```

Comment:
We shouldn't recommend passing all data about the record in the URL. `id` should be enough and we can put the record data to the state, then check it on the route resolution stage and load the data by `id` if the state is empty.

Reply:
And, based on the discussion we had, we can have a method to handle `id`.